### PR TITLE
Make completion status work for dynamicModalOverlays

### PR DESCRIPTION
**Pull request by ntsummers1**
### Description

Make the completion status show that green checkmark thingy that gives the happy feels instead of the red exclaimation thingy that makes you sad

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/34715f60-04de-4056-8d0f-5705c6b50b8e

Things I'd like to do at a later time

- Fix up how completionStatus matches fieldData to form templates for initiatives. Too many for loops. Way too complex.
- Validation for choicelists in the initiatives isn't showing validation correctly
- Breakup evaluationPlan and fundingSources extraction to separate functions for cleaner tests
- Write more tests for the different objective card scenarios (what data has to show and when)
- Remove the unused data we don't need from workplan thats being pass to the SAR
- Confirm whether completionStatus needs to see at least 4 types of initiatives in order to submit or not

### Related ticket(s)

CMDCT-3175

---
### How to test

Create a SAR
Check review and submit page to see its not completed
Fill out the State and Territory Objectives page
See the green checkmark appear!

---
### Author checklist

- [x] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [x] I have updated relevant documentation, if necessary

### CODE REVIEW

**Review by benmartin-coforma — APPROVED** [review]
Very nice!

File: services/app-api/utils/validation/completionStatus.ts [hunk 1]
```diff
@@ -170,6 +169,20 @@ export const calculateCompletionStatus = async (
...
+                    ? stepForm.form
+                    : stepForm.modalForm;
+                  const isEntityComplete = await calculateFormCompletion(
+                    nestedFormTemplate,
+                    stepFields
+                  );
+                  areAllFormsComplete &&= isEntityComplete;
+                }
+              }
+            }
```
[the remark this answers is not in the document]
Comment (ntsummers1) [reply services/app-api/utils/validation/completionStatus.ts:194]: Added!

**Review by cassandradanger — COMMENTED** [review]
dunno if this is just a me issue? But i filled out all the state initiative stuff - seeing green on the initiative dashboard, but not on the review and submit page
Screenshot 2024-01-25 at 10 33 32 AM
Screenshot 2024-01-25 at 10 33 43 AM

**Comment by ailZhou** [comment]
same issue as @cassandradanger, state initiative is still showing false on the review and submit page even when all the initiatives have been filled out

**Comment by ntsummers1** [comment]
Please forgive me. I didn't push up this swap of code 😰

**Review by cassandradanger — COMMENTED** [review]
looks like we lost some answers showing in objectives, and styling
current:
Screenshot 2024-01-29 at 10 50 31 AM

expected:
Screenshot 2024-01-29 at 10 55 49 AM

**Comment by ailZhou** [comment]
Pro: was able to finally get a green checkmark and submit the form 🥳 !
Con:  Same styling issue as @cassandradanger but there seems to be a bug with the edit not clearing out previous child data. Probably can be addressed in a separate PR 🤔.

Screenshot 2024-01-29 at 12 00 22 PM

**Review by ailZhou — APPROVED** [review]
🚀

**Pull request merged**